PR: ACES 2.0

KelSolaar opened the pull request:
Starting a PR for the ACES 2.0 work, I haven't even tried to build the configs but the classifying code is working, so did a solid good pass at updating the spreadsheets:

- *Reference*: https://docs.google.com/spreadsheets/d/1z3xsy3sF0I-8AN_tkMOEjHlAs13ba7VAVhrE8v4WIyo/edit?usp=sharing
- *CG & Studio*: https://docs.google.com/spreadsheets/d/1PXjTzBVYonVFIceGkLDaqcEJvKR6OI63DwZX0aajl3A/edit?usp=sharing

Current build artifacts can be found here:

### Reference

- [reference-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19179983/reference-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt)

### CG

- [cg-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19224479/cg-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt)

### Studio

- [studio-config-all-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19180003/studio-config-all-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt)
- [studio-config-d60-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19180004/studio-config-d60-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt)
- [studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19180005/studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt)

KelSolaar commented:
First artifacts are here: https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/actions/runs/10648206791

I haven't checked anything yet.

carolalynn commented:
Amazing Thomas - just want to note a few quick things:

CG Config:
- We had a few more views / displays in previous versions:
    - PQ P3 1000 nit Rec. 2020 lim
    - PQ P3 1000 nit P3 lim
    - SDR Cinema

Studio config:
- Name of look transforms seem off, the gamut compression is named "ACES2065-1"
- Did we decide to split the D65 / D60 into two configs?

General:
- talk about "extended" displays for Display P3 etc? Might want these in both configs.
- will do a more detailed naming/alias check later once CIF decides on names

doug-walker commented:
I added three of the missing display built-in transforms to your spreadsheet (only one still missing).

Looking at the color spaces, I'm noticing both "Apple Log" and "AppleLog BT2020", despite the latter not being in the spreadsheet.

KelSolaar commented:
I implemented support for:

- Passing the encoding as a suffix to the DisplayColorspace name
- Excluding the D60 views although we need to discuss how we want to name the Configs

doug-walker commented:
@KelSolaar , thanks for adding the ability to have an SDR and HDR version of DisplayP3!  Here are some comments looking at the most recent Studio config artifact:

- I like your use of the "SDR - Display" and "HDR - Display" suffixes on DisplayP3. However, I'm wondering if some people will complain that some display names indicate range and others don't. An alternative would be to leave the SDR displays as "Display" and change *all* the HDR displays to "HDR Display". But on the other hand, indicating the range on just those displays that have both could be considered a benefit. Minor point, but do we need the second dash, perhaps "Display P3 - HDR Display" would be sufficient?
- If we do use the proposed "Display P3 - SDR - Display" name, we should add the previous name and "displayp3_display" aliases for backwards compatibility.
- It looks like for all the HDR displays, we are include the SDR ACES views, to have them available for soft-proofing. So it's expected to see the SDR displays in the HDR version. However, please remove the HDR views from the SDR display.
- This config continues to have both the D65 and D60 views (I guess the new option to remove D60 was not in effect?). However, the Display P3 views are a strange and incomplete mix of D60 and D65.
- Both display color spaces have the same set of ACEStransformIDs. Not sure what to do here. Probably we should only include the SDR views with the SDR display, and HDR views with the HDR display. But just wondering if there would be scenarios where it's safer to have all of them?

doug-walker commented:
I had an action-item from the last meeting to check the ACES v2 config to see if we have all the color spaces from the ACES v1.3 configs. Here are my findings:
- For the named_transforms, "ARRI LogC3 - Curve (EI800)" is missing.
- All scene-referred color spaces are present.
- For the display-referred color spaces, the displays that are not included in the ACES v2 set are missing, these are: "Rec.1886 Rec.2020 - Display", "P3-D60 - Display", and "P3-DCI - Display". 

Not sure what to do about the displays. On the one hand, they are not very relevant and I'm open to dropping them. On the other hand, I'm just wondering if anyone will complain about missing displays when moving from ACES v1 to v2.

KelSolaar commented:
Hello,

Thanks for the thorough review!

### 1
> Minor point, but do we need the second dash, perhaps "Display P3 - HDR Display" would be sufficient?

The second dash is coming from the `- Display` suffix that we consistently add to all the display colourspaces. It feels a bit arbitrary to add that to Display P3 when none of the other HDR display has it. What about `Display P3 SDR - Display` and `Display P3 HDR - Display`?

### 2
> If we do use the proposed "Display P3 - SDR - Display" name, we should add the previous name and "displayp3_display" aliases for backwards compatibility.

Yes, good catch!

### 3

> It looks like for all the HDR displays, we are include the SDR ACES views, to have them available for soft-proofing. So it's expected to see the SDR displays in the HDR version. However, please remove the HDR views from the SDR display.

Where did you see this? The only instances I saw are occurring for Display P3 which is "expected" because we use the same builtin for both HDR and SDR. It would be much cleaner to have a dedicated *Display P3 HDR* builtin as suggested in previous meetings:

- I would not need to use that new gross encoding column that we added for Display P3
- The generator would work perfectly fine as it was intended

I'm happy to make a PR for this on the OCIO side.

### 4

> I guess the new option to remove D60 was not in effect?

Yes, I will check what happened for Display P3, maybe the quick filterer I wrote is not "smart" enough or there are yet another problem because we introduced that special case as mentioned earlier.

### 5

> Both display color spaces have the same set of ACEStransformIDs. Not sure what to do here.

It might be time to source this AMF data from the ACES side and stop generating it on our end.

### 6
> For the named_transforms, "ARRI LogC3 - Curve (EI800)" is missing.

I will check, it might not even be in the spreadsheet.

### 7

> the displays that are not included in the ACES v2 set are missing

If they are not in ACES 2.0, they cannot be in the config at the moment.

### 8
> There are a lot of aliases that are missing.

If you have a list we could talk about it yep!

doug-walker commented:
Following up on our working group meeting today, here is the list of missing aliases, comparing the latest ACES 2 studio config to the current OCIO 2.4 built-in studio config. As I mentioned, we could consider dropping some of them, but here's the list of all of them (color spaces and named transforms). Thomas, please let me know if you want me to add any of these to the spreadsheet.

ACEScg
-- Missing alias:  lin_ap1_scene
Linear ARRI Wide Gamut 3
-- Missing alias:  lin_alexawide 
ARRI LogC3 (EI800)
-- Missing alias:  logc3ei800_alexawide
CanonLog2 CinemaGamut D55
-- Missing alias:  canonlog2_cgamutday
CanonLog3 CinemaGamut D55
-- Missing alias:  canonlog3_cgamutday
Linear CinemaGamut D55
-- Missing alias:  lin_canoncgamutday
Linear REDWideGamutRGB
-- Missing alias:  lin_rwg
Log3G10 REDWideGamutRGB
-- Missing alias:  rl3g10_rwg
S-Log3 S-Gamut3.Cine
-- Missing alias:  slog3_sgamutcine
S-Log3 Venice S-Gamut3.Cine
-- Missing alias:  slog3_venice_sgamutcine
sRGB Encoded Rec.709 (sRGB)
-- Missing alias:  srgb_texture
-- Missing alias:  srgb_rec709_scene
-- Missing alias:  Input - Generic - sRGB - Texture
-- Missing alias:  sRGB - Texture
-- Missing alias:  srgb_tx
Gamma 1.8 Encoded Rec.709
-- Missing alias:  g18_rec709
-- Missing alias:  Gamma 1.8 Rec.709 - Texture
-- Missing alias:  g18_rec709_tx
-- Missing alias:  g18_rec709_scene
Gamma 2.2 Encoded Rec.709
-- Missing alias:  g22_rec709
-- Missing alias:  Gamma 2.2 Rec.709 - Texture
-- Missing alias:  g22_rec709_tx
-- Missing alias:  g22_rec709_scene
Camera Rec.709
-- Missing alias:  rec709_camera
sRGB Encoded P3-D65
-- Missing alias:  srgb_displayp3_scene
-- Missing alias:  sRGB Encoded P3-D65 - Texture
-- Missing alias:  srgb_encoded_p3d65_tx
Linear Rec.709 (sRGB)
-- Missing alias:  lin_rec709
-- Missing alias:  lin_rec709_scene
-- Missing alias:  lin_srgb
-- Missing alias:  Utility - Linear - sRGB
Linear P3-D65
-- Missing alias:  lin_displayp3
-- Missing alias:  lin_displayp3_scene
-- Missing alias:  Linear Display P3
Linear AdobeRGB
-- Missing alias:  lin_adobergb_scene
Linear Rec.2020
-- Missing alias:  lin_rec2020_scene

C-Log2 - Curve
-- Missing alias:  crv_canonlog2
C-Log3 - Curve
-- Missing alias:  crv_canonlog3
V-Log - Curve
-- Missing alias:  crv_vlog
Log3G10 - Curve
-- Missing alias:  crv_rl3g10
S-Log3 - Curve
-- Missing alias:  crv_slog3
sRGB - Curve
-- Missing alias:  crv_srgb
Rec.1886 - Curve
-- Missing alias:  crv_rec1886
Rec.709 - Curve
-- Missing alias:  crv_rec709

doug-walker commented:
@KelSolaar , I looked again at the alias differences I reported above and found a bug in my comparison script. The changes needed are actually very minimal.  Please simply add "lin_ap1_scene" to ACEScg.

doug-walker commented:
@KelSolaar , if you would like to add the additional DisplayP3 HDR built-in, you would make a second call to [registry.addBuiltin here](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/9bc5b8eb1c365b0b1a4fa66e1f3a54646b746120/src/OpenColorIO/transforms/builtins/Displays.cpp#L358).

Suggested name: "DISPLAY - CIE-XYZ-D65_to_DisplayP3-HDR", but I'm flexible.

KelSolaar commented:
OP updated with the latest config artifacts!

doug-walker commented:
Found a bug:  The Rec.2100-PQ Display does not have the view "ACES 2.0 - HDR 4000 nits (P3 D65)", which it is supposed to have. Applies to both the D65 and All versions.

doug-walker commented:
Another bug: Compared to the most recent ACES 1.3 config, the new config has these added color spaces:
- "AppleLog BT2020" (this is a duplicate of "Apple Log")
- "Log3G10 RWG" (this is a duplicate of "Log3G10 REDWideGamutRGB")

@KelSolaar , I don't see these in the spreadsheet, where are they coming from?

KelSolaar commented:
Most likely: The Reference Config spreadsheet, we do a kind of union of both.

doug-walker commented:
As I mentioned during the TSC meeting, I would like to re-order the color spaces so that they are grouped together and appear in a more logical order in menus that are not hierarchical by family. I was planning on doing this by adjusting the values in the Ordering column of the spreadsheet. However, it does not seem like this is currently being followed. For example, "V-Log V-Gamut" has an Ordering of 130, yet it appears above "BMDFilm WideGamut Gen5" which has an Ordering of 115. 

@KelSolaar, What am I missing, how should I do the re-ordering?

KelSolaar commented:
> Found a bug: The Rec.2100-PQ Display does not have the view "ACES 2.0 - HDR 4000 nits (P3 D65)", which it is supposed to have. Applies to both the D65 and All versions.

There was an error in the spreadsheet: The incorrect display was bound.

> @KelSolaar , I don't see these in the spreadsheet, where are they coming from?

Confirming that they were coming from the Reference Config spreadsheet

> @KelSolaar, What am I missing, how should I do the re-ordering?

It is kind of similar to previous point, those are coming from the Reference Config spreadsheet so their order there will take precedence. A dirty option could be to not include them from the Reference Config spreadsheet and I think that things should work. Ultimately, I think we should find a way to merge the Reference and CG / Studio Config spreadsheets. This would provide an holistic understanding of what is going on. If it was easy though, I would have done it :)

KelSolaar commented:
Artifacts updated!

KelSolaar commented:
The order of the displays and view is driven by the `Ordering` column in that spreadsheet: https://docs.google.com/spreadsheets/d/1z3xsy3sF0I-8AN_tkMOEjHlAs13ba7VAVhrE8v4WIyo/edit?gid=0#gid=0 Is there a particular ordering we would like?

KelSolaar commented:
I just push a code change that should allow maintaining the ordering defined in the *CG* and *Studio* configs irrespective of what it was first in the *Reference* config.

doug-walker commented:
Thank you for the updates @KelSolaar!  Here is a strawman for the display/view orderings to address the excellent point raised by @mjtitchener-fn.  Any thoughts/comments?

Active displays I made alphabetical and then moved sRGB to the top. The alphabetical ordering is almost SDR, then HDR, which is convenient. The config.getDefaultDisplay() function will return sRGB.

The Active views ordering is only relevant within the views supported by a given display, so the Rec.2100-PQ is the most challenging since it's the only one that has both HDR and SDR views. For this reason, I put the HDR views first, starting with 1000 nits, so that config.getDefaultView("Rec.2100-PQ - Display") returns "ACES 2.0 - HDR 1000 nits (P3 D65)".

(By the way, the active display/view lists is optional, so we could just use the ordering within the display and view lists themselves, if that is preferable for some reason.)

active_displays: [
sRGB - Display
Display P3 - Display
Display P3 HDR - Display]
P3-D65 - Display
Rec.1886 Rec.709 - Display
Rec.2100-HLG - Display
Rec.2100-PQ - Display
ST2084-P3-D65 - Display

active_views: [
ACES 2.0 - HDR 1000 nits (P3 D65)
ACES 2.0 - HDR 1000 nits (Rec.2020)
ACES 2.0 - HDR 2000 nits (P3 D65)
ACES 2.0 - HDR 2000 nits (Rec.2020)
ACES 2.0 - HDR 4000 nits (P3 D65)
ACES 2.0 - HDR 4000 nits (Rec.2020)
ACES 2.0 - HDR 500 nits (P3 D65)
ACES 2.0 - HDR 500 nits (Rec.2020)
ACES 2.0 - HDR 108 nits (P3 D65)
ACES 2.0 - SDR 100 nits (Rec.709)
ACES 2.0 - SDR 100 nits (P3 D65)
Un-tone-mapped
Raw]

KelSolaar commented:
After spending quite a bit of time on this, I cannot figure out an elegant way to handle all the peculiarities of the desired ordering: What determines the order of the `displays`, `active_displays` and `active_views` is the order in which the `shared_views` are generated by the code which is driven by the spreadsheet and now the second display related ordering pass. I don't really see at the moment how we can have `sRGB - Display` first and thus `ACES 2.0 - SDR 100 nits (Rec.709)` first in the `shared_views` but have towards the end of the `active_views` without resorting to some sort of hack in the code.

I will attach update the OP with the current configs for discussion.

doug-walker commented:
Hey Thomas, thanks for the updates. I will confirm at the Wednesday meeting, but it does seem mandatory that the 1000nit view needs to be highest priority for the HDR displays. It also seems mandatory that the sRGB display is the default. Maybe we could just add a hack to move the sRGB display to the front of the active_displays list?

Do I understand your earlier comment that the ordering of the Studio config is now solely driven by the Studio spreadsheet? I'm not seeing that in the config I looked at, the D65 one. For example, the spreadsheet has:
440  P3-D65_500nit_in_Rec2100-D65_ST2084
441  P3-D65_1000nit_in_Rec2100-D65_ST2084
And yet the 1000nit comes first in the config.

Another example for color spaces:
110  Linear ARRI Wide Gamut 3
111  ARRI LogC4
And yet LogC4 comes first in the config.

KelSolaar commented:
The ordering of the displays and views only honours the spreadsheet partially as per above and explains the current config state which I think is close to what we want except for the active views?:

```yaml
shared_views:
  - !<View> {name: ACES 2.0 - SDR 100 nits (Rec.709), view_transform: ACES 2.0 - SDR 100 nits (Rec.709), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), view_transform: ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - SDR 100 nits (P3 D65), view_transform: ACES 2.0 - SDR 100 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 1000 nits (P3 D65), view_transform: ACES 2.0 - HDR 1000 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), view_transform: ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 500 nits (P3 D65), view_transform: ACES 2.0 - HDR 500 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 2000 nits (P3 D65), view_transform: ACES 2.0 - HDR 2000 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 4000 nits (P3 D65), view_transform: ACES 2.0 - HDR 4000 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 500 nits (Rec.2020), view_transform: ACES 2.0 - HDR 500 nits (Rec.2020), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 1000 nits (Rec.2020), view_transform: ACES 2.0 - HDR 1000 nits (Rec.2020), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 2000 nits (Rec.2020), view_transform: ACES 2.0 - HDR 2000 nits (Rec.2020), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 4000 nits (Rec.2020), view_transform: ACES 2.0 - HDR 4000 nits (Rec.2020), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), view_transform: ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), view_transform: ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 108 nits (P3 D65), view_transform: ACES 2.0 - HDR 108 nits (P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), view_transform: ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), display_colorspace: <USE_DISPLAY_NAME>}
  - !<View> {name: Un-tone-mapped, view_transform: Un-tone-mapped, display_colorspace: <USE_DISPLAY_NAME>}

displays:
  sRGB - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), Un-tone-mapped]
  Display P3 - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), Un-tone-mapped]
  Display P3 HDR - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), Un-tone-mapped]
  P3-D65 - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), Un-tone-mapped]
  Rec.1886 Rec.709 - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), Un-tone-mapped]
  Rec.2100-HLG - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), Un-tone-mapped]
  Rec.2100-PQ - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), Un-tone-mapped]
  ST2084-P3-D65 - Display:
    - !<View> {name: Raw, colorspace: Raw}
    - !<Views> [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), Un-tone-mapped]

active_displays: [sRGB - Display, Display P3 - Display, Display P3 HDR - Display, P3-D65 - Display, Rec.1886 Rec.709 - Display, Rec.2100-HLG - Display, Rec.2100-PQ - Display, ST2084-P3-D65 - Display]
active_views: [ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), Un-tone-mapped, Raw]
```

I will add a 

The colorspaces are ordered according to the spreadsheet but basically what is happening here is that we have some colorspaces that are internally doubled because of their Reference config variant and their Studio / CG config CLF variant, e.g., `Apple Log`, `ARRI LogC4` so I will do a bit of cleanup:

![image](https://github.com/user-attachments/assets/140685d3-a3bd-4bbb-8284-2bc3e5f9681b)

KelSolaar commented:
Btw, for more context, here is how the views are ordered internally:

![image](https://github.com/user-attachments/assets/cd8704a2-b558-4be1-b003-a6e6e4186672)

```json
{
  "shared_views": [
      {
        "display": "sRGB - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709)"
      },
      {
        "display": "sRGB - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)"
      },
      {
        "display": "Display P3 - Display",
        "view": "ACES 2.0 - SDR 100 nits (P3 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (P3 D65)"
      },
      {
        "display": "Display P3 - Display",
        "view": "ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "Display P3 HDR - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D65)"
      },
      {
        "display": "Display P3 HDR - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "P3-D65 - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709)"
      },
      {
        "display": "P3-D65 - Display",
        "view": "ACES 2.0 - SDR 100 nits (P3 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (P3 D65)"
      },
      {
        "display": "P3-D65 - Display",
        "view": "ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "P3-D65 - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65)"
      },
      {
        "display": "Rec.1886 Rec.709 - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709)"
      },
      {
        "display": "Rec.1886 Rec.709 - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)"
      },
      {
        "display": "Rec.2100-HLG - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D65)"
      },
      {
        "display": "Rec.2100-HLG - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 500 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 500 nits (P3 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 2000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (P3 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 4000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (P3 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 500 nits (Rec.2020)",
        "view_transform": "ACES 2.0 - HDR 500 nits (Rec.2020)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 1000 nits (Rec.2020)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (Rec.2020)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 2000 nits (Rec.2020)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (Rec.2020)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 4000 nits (Rec.2020)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (Rec.2020)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65)"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65)",
        "view_transform": "ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 500 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 500 nits (P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 2000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 4000 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 108 nits (P3 D65)",
        "view_transform": "ACES 2.0 - HDR 108 nits (P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65)",
        "view_transform": "ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65)"
      },
      {
        "display": "sRGB - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "Display P3 - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "Display P3 HDR - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "P3-D65 - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "Rec.1886 Rec.709 - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "Rec.2100-HLG - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "Rec.2100-PQ - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      },
      {
        "display": "ST2084-P3-D65 - Display",
        "view": "Un-tone-mapped",
        "view_transform": "Un-tone-mapped"
      }
    ]
}
```

doug-walker commented:
Thanks Thomas! Yes, the active_displays now seem fine, so if the active_views could be reordered to put HDR 1000 nits first, that would be helpful. I validated this with the group during the Wednesday ACES 2 call.

In addition, the group agreed that we want to add an SDR view to all the HDR displays, not only the Rec.2100 PQ one (since it's so helpful to be able to preview how things would look on the most restricted range/gamut device). The request is that the "ACES 2.0 - SDR 100 nits (Rec.709)" view get added to these displays:
Display P3 HDR - Display
Rec.2100-HLG - Display
ST2084-P3-D65 - Display

I realize this is not following the CTL, so how difficult a change is it?

KelSolaar commented:
> I realize this is not following the CTL, so how difficult a change is it?

Ain't not CTL ain't no cigar unfortunately without excessive and unmaintainable hacking! We should add them there if we want them.

KelSolaar commented:
Op updated with latest and greatest, I think this addresses all the last comments but the new view request.

carolalynn commented:
Amazing Thomas - I’ll also take a look tomorrow.

On Wed, Feb 26, 2025 at 17:45 Thomas Mansencal ***@***.***>
wrote:

> Op updated with latest and greatest, I think this addresses all the last
> comments but the new view request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2686596346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMPHU2GDISCFEWUJM37NHQ32RZU3XAVCNFSM6AAAAABL5SWWTSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOBWGU4TMMZUGY>
> .
> You are receiving this because you commented.Message ID:
> <AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130/c2686596346@
> github.com>
> [image: KelSolaar]*KelSolaar* left a comment
> (AcademySoftwareFoundation/OpenColorIO-Config-ACES#130)
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2686596346>
>
> Op updated with latest and greatest, I think this addresses all the last
> comments but the new view request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2686596346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMPHU2GDISCFEWUJM37NHQ32RZU3XAVCNFSM6AAAAABL5SWWTSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOBWGU4TMMZUGY>
> .
> You are receiving this because you commented.Message ID:
> <AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130/c2686596346@
> github.com>
>

KelSolaar commented:
Thanks a ton @scottdyer!

I haven't checked but I pulled your PR into [our fork of aces-system](https://github.com/colour-science/aces-system), updated the spreadsheet, etc... Doing so I noticed that some of the encodings were incorrect for some views. Anyway, @doug-walker, @carolalynn, OP configs are updated with those 4 new views, HLG is not included as per https://github.com/ampas/aces-output/pull/11/commits/cee5815d3279e28f44e3dfe7231a20773f6e75af

carolalynn commented:
Thanks Thomas - do we need the inverses specifically in the CTL on our
side? They won't really be correct / pass the cube image test. We were
discussing on slack whether it made sense to have them at all in the CTL.

On Fri, Feb 28, 2025 at 11:10 AM Thomas Mansencal ***@***.***>
wrote:

> Thanks a ton @scottdyer <https://github.com/scottdyer>!
>
> I haven't checked but I pulled your PR into our fork of aces-system
> <https://github.com/colour-science/aces-system>, updated the spreadsheet,
> etc... Doing so I noticed that some of the encodings were incorrect for
> some views. Anyway, @doug-walker <https://github.com/doug-walker>,
> @carolalynn <https://github.com/carolalynn>, OP configs are updated with
> those 4 new views, HLG is not included as per ***@***.***
> <https://github.com/ampas/aces-output/commit/cee5815d3279e28f44e3dfe7231a20773f6e75af>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2691366872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMPHU2HBD75GSPZPI4QYGLT2SCYATAVCNFSM6AAAAABL5SWWTSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM3DMOBXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> <AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130/c2691366872@
> github.com>
> [image: KelSolaar]*KelSolaar* left a comment
> (AcademySoftwareFoundation/OpenColorIO-Config-ACES#130)
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2691366872>
>
> Thanks a ton @scottdyer <https://github.com/scottdyer>!
>
> I haven't checked but I pulled your PR into our fork of aces-system
> <https://github.com/colour-science/aces-system>, updated the spreadsheet,
> etc... Doing so I noticed that some of the encodings were incorrect for
> some views. Anyway, @doug-walker <https://github.com/doug-walker>,
> @carolalynn <https://github.com/carolalynn>, OP configs are updated with
> those 4 new views, HLG is not included as per ***@***.***
> <https://github.com/ampas/aces-output/commit/cee5815d3279e28f44e3dfe7231a20773f6e75af>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2691366872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMPHU2HBD75GSPZPI4QYGLT2SCYATAVCNFSM6AAAAABL5SWWTSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJRGM3DMOBXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> <AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130/c2691366872@
> github.com>
>

scottdyer commented:
> Doing so I noticed that some of the encodings were incorrect for some views.
Do you know which ones? Some of the newly added views or ones that previously existed? 
I would like to fix them quickly if they're broken.

KelSolaar commented:
Sorry for the confusion @scottdyer, this was in our spreadsheets!

doug-walker commented:
Hi Thomas, thanks for the updates. I see the SDR view in the "ST2084-P3-D65 - Display" but not in the "Display P3 HDR - Display". However, I noticed that the "Display P3 - Display" now has both Rec.709 and P3 views. The CTL only has a P3 view. I'm guessing what happened is the new view got added to the wrong Display P3 display. 

That said, I'm wondering if for "Display P3 HDR - Display" we want the Rec.709 or P3 SDR view? The latter might make more sense for people wanting to compare what the SDR view would look like.  

One use-case is of wanting to look at the most restricted view, which would be Rec.709. However, it might be more useful to see the SDR P3 view on "Display P3 HDR- Display" since that is more likely what they would get. Maybe that's true for the "ST2084-P3-D65 - Display" too?

And I'm noticing the active_views lists are not updated in this drop, I think that one remains on your todo list, but wanted to confirm.  Thanks!

KelSolaar commented:
>  I see the SDR view in the "ST2084-P3-D65 - Display" but not in the "Display P3 HDR - Display". However, I noticed that the "Display P3 - Display" now has both Rec.709 and P3 views. The CTL only has a P3 view. I'm guessing what happened is the new view got added to the wrong Display P3 display.

I did not put them in the correct display in the spreadsheet! :(

> And I'm noticing the active_views lists are not updated in this drop, I think that one remains on your todo list, but wanted to confirm. Thanks!

The code has not changed at all since my last update, the `active_views` looks like this for me, it is different for you?

```yaml
active_views: [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), Un-tone-mapped, Raw]
```

KelSolaar commented:
Op updated with views moved to `DisplayP3 - HDR`.

mjtitchener-fn commented:
@KelSolaar Thanks for all of your effort on this (and in general)!

The defaults for `active_displays` and `active_views` look good to me now.

> The code has not changed at all since my last update, the active_views looks like this for me, it is different for you?

This made me wonder which config everyone is testing/referring to? I've mainly been using [studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19048227/studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt) as I _believe_ this is the one we'll be promoting as the "_main_" config for use/inclusion in DCCs? 

The `active_views` example in [this](https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/pull/130#issuecomment-2693350326) post from @KelSolaar though is from [studio-config-all-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt](https://github.com/user-attachments/files/19048225/studio-config-all-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio.txt).

Both are valid of course but it could result in confusion if we're looking at different things.

KelSolaar commented:
I think that I finally figured out what is happening in the depth of the Reference config generator and producing unpredictable ordering. I'll do more tests.

KelSolaar commented:
Ok, OP updated, involved deep and dangerous changes to the reference config generator that everything depends on, hopefully, nothing broke. It is extremely painful not being able to test this with confidence. @mjtitchener-fn : You are my eyes here, let me know! :)

KelSolaar commented:
Thanks for checking, one step forward, i.e., things are ordered properly but two backwards, i.e., there is now duplication (which is why I was grouping the spreadsheet elements in a particular way before when building the internal data).

This should be an easy fix and I have ocioview finally running now, so it will be easier to test!

KelSolaar commented:
@mjtitchener-fn: I did a one liner change that should avoid duplication, cursory check of the internal data and in OCIO view seems to show that things are trending in the correct direction. Let me know, I need to jump on the other job.

mjtitchener-fn commented:
Thanks @KelSolaar, yep this has fixed the duplication problem.
For clarity here’s what I’m seeing as the default view for each display.
This is using `studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio`

**sRGB - Display**
ACES 2.0 - SDR 100 nits (Rec.709)

**Display P3 - Display**
ACES 2.0 - SDR 100 nits (P3 D65)

**Display P3 HDR - Display**
ACES 2.0 - HDR 1000 nits (P3 D65)

**P3-D65 - Display**
ACES 2.0 - SDR 100 nits (Rec.709)
_Should this default to ACES 2.0 - SDR 100 nits (P3 D65) instead?_

**Rec.1886 Rec.709 - Display**
ACES 2.0 - SDR 100 nits (Rec.709)*

**Rec.2100-HLG - Display**
ACES 2.0 - HDR 1000 nits (P3 D65)
_Only a P3 D65 view is available for HLG. Given this is Rec.2100 shouldn't a Rec.2020 view be available and default?_ 

**Rec.2100-PQ - Display**
ACES 2.0 - HDR 500 nits (P3 D65)
_As above if this is Rec.2100 should a Rec.2020 view be default (they are avaiable for this display). Also would 1000 nits be preferential as a default?_

**ST2084-P3-D65 - Display**
ACES 2.0 - HDR 108 nits (P3 D65)
_I would've thought this should default to ACES 2.0 - HDR 500 nits (P3 D65) or ACES 2.0 - HDR 1000 nits (P3 D65)?_

I know it’s hard to have it all ways when setting rules to do to this programitically and users can ultimately select what's right for their workflow. Just thinking out loud about what the best starting points are 🤔

doug-walker commented:
@mjtitchener-fn , thanks for testing this!  Regarding HLG, the ACES 2 working group only made the 1000 nit P3 flavor available. My impression is that P3 is the preferred default across the different HDR displays since with Rec.2020 one cannot see it yet (except with various exotic displays) and so there is not the ability to confidently master to that yet.  I would make the 1000 nit P3 the default for those last three items in your list.

doug-walker commented:
@KelSolaar , per my action item from today's meeting, please make the following changes to the active_views lists. The rest of the configs look mostly ok.  With the following exceptions:

1. The AMF data in the color space descriptions went haywire, and there is a lot of extra incorrect stuff now. If you are able to easily turn off the AMF descriptions for now, that would be great. Otherwise it's ok, though the volume of stuff is large, especially in the D60 config.
2. The D60 config is missing the Raw and Un-tone-mapped views. If it's difficult to add these easily, please remove this config for now.
3. For the CG config, could you remove the D60 and All versions from the generator?  If not, let me know and I'll do the active_views for those.
4. As mentioned previously, the "P3-D65 - Display" and "DCDM-D65 - Display" are using "100 nits" views, despite the name having "48 nits" in the CTLs. 

Active view list for config: studio-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio
`active_views: [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (Rec.709), Un-tone-mapped, Raw]`

The default views will then be as follows:
```
sRGB - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Display P3 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Display P3 HDR - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
P3-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Rec.1886 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Rec.2100-HLG - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
Rec.2100-PQ - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
ST2084-P3-D65 - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
```

Active view list for config: studio-config-d60-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio
`active_views: [ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), Un-tone-mapped, Raw]`

The default views will then be as follows:
```
sRGB - Display :   ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)
Display P3 - Display :   ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)
Display P3 HDR - Display :   ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)
P3-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65)
Rec.1886 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65)
Rec.2100-HLG - Display :   ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)
Rec.2100-PQ - Display :   ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65)
ST2084-P3-D65 - Display :   ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65)
```

Active view list for config: studio-config-all-views-v2.3.0_aces-v2.0_ocio-v2.4.ocio
`active_views: [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), Un-tone-mapped, Raw]`

The default views will then be as follows:
```
sRGB - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Display P3 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Display P3 HDR - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
P3-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Rec.1886 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Rec.2100-HLG - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
Rec.2100-PQ - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
ST2084-P3-D65 - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
```

Active view list for config: cg-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio
`active_views: [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (Rec.709), Un-tone-mapped, Raw]`

The default views will then be as follows:
```
sRGB - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Display P3 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Display P3 HDR - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
P3-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Rec.1886 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Rec.2100-PQ - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
```

Active view list for config: reference-config-v2.3.0_aces-v2.0_ocio-v2.4.ocio
active_views: [ACES 2.0 - HDR 1000 nits (P3 D65), ACES 2.0 - HDR 1000 nits (Rec.2020), ACES 2.0 - HDR 1000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 1000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 1000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (P3 D65), ACES 2.0 - HDR 2000 nits (Rec.2020), ACES 2.0 - HDR 2000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 2000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 2000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (P3 D65), ACES 2.0 - HDR 4000 nits (Rec.2020), ACES 2.0 - HDR 4000 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 4000 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 4000 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (P3 D65), ACES 2.0 - HDR 500 nits (Rec.2020), ACES 2.0 - HDR 500 nits (P3 D60 in P3 D65), ACES 2.0 - HDR 500 nits (P3 D60 in Rec.2020 D65), ACES 2.0 - HDR 500 nits (Rec.2020 D60 in Rec.2020 D65), ACES 2.0 - HDR 300 nits (P3 D65), ACES 2.0 - HDR 300 nits (P3 D60 in XYZ E), ACES 2.0 - HDR 108 nits (P3 D65), ACES 2.0 - HDR 108 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (P3 D65), ACES 2.0 - SDR 100 nits (P3 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709), ACES 2.0 - SDR 100 nits (Rec.709 D60 in P3 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.2020 D65), ACES 2.0 - SDR 100 nits (Rec.709 D60 in Rec.709 D65), ACES 2.0 - SDR 100 nits (P3 D60 in XYZ E), Un-tone-mapped, Raw]

The default views will then be as follows:
```
sRGB - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
DCDM-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Display P3 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Display P3 HDR - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
G2.2 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
P3-D65 - Display :   ACES 2.0 - SDR 100 nits (P3 D65)
Rec.1886 Rec.709 - Display :   ACES 2.0 - SDR 100 nits (Rec.709)
Rec.2100-HLG - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
Rec.2100-PQ - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
ST2084-DCDM-D65 - Display :   ACES 2.0 - HDR 300 nits (P3 D65)
ST2084-P3-D65 - Display :   ACES 2.0 - HDR 1000 nits (P3 D65)
```

KelSolaar commented:
Thanks @doug-walker!

1. As we talk quite a few times, the AMF components should really come from somewhere else than the Config generator. There should be a file / database somewhere on the Academy side that we can pull from as required. Disabled for now!
2. Good catch, and `/done`.
3. Easy! `/done`.
4. The View name is programmatically derived from the `BuiltinTransform`, and there is not `48nit` one, not much I can do unfortunately!

``` python
>>> [k for k in opencolorio_config_aces.BUILTIN_TRANSFORMS if "48nit" in k]
[]
>>> [k for k in opencolorio_config_aces.BUILTIN_TRANSFORMS if "100nit" in k]
['ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-REC709_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-P3-D65_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-REC709-D60-in-REC709-D65_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-REC709-D60-in-P3-D65_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-REC709-D60-in-REC2020-D65_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-P3-D60-in-P3-D65_2.0', 'ACES-OUTPUT - ACES2065-1_to_CIE-XYZ-D65 - SDR-100nit-P3-D60-in-XYZ-E_2.0']
```

I will work on the active views ordering now.

KelSolaar commented:
OP updated, the ordering should match for most of the configs but:

- There are some minor differences for the D60 variant, its is currently the result from the filtering all the other views so maintains the order of the `all-views`.
- The CG config has less displays, I'm not sure why we had the 500, 2000 and 4000 nits displays.

I fixed an error introduced whilst doing the Reference config changes for the ordering, this should have sorted most of the descriptions and the AMF components I think.

doug-walker commented:
Thanks for the update @KelSolaar !  The Reference and Studio configs now look fine to me. Unfortunately, I made an error when I reviewed the CG config.  For ACES 1.3 we only included the HDR 1000 nit view and our displays list included "ST2084-P3-D65 - Display".  I think we should continue likewise for the ACES 2 configs.  You already fixed my error of including the non-1000 nit HDR views, so the active_views list is correct. Please just add the "ST2084-P3-D65 - Display" display to the CG config and we should be good to release.

KelSolaar commented:
*CG* config is updated, I added `P3-D65_1000nit_in_P3-D65_ST2084` to it.

doug-walker commented:
Thank you @KelSolaar, all issues now seem to have been addressed.  Please proceed with the release!!   :)

doug-walker reviewed (approved):
LGTM!  Awesome work Thomas!
